tools/mtmd/clip.cpp

```diff
@@ -8,15 +8,19 @@
 #include "ggml-cpp.h"
 #include "ggml-alloc.h"
 #include "ggml-backend.h"
+#include "ggml/src/ggml-impl.h"
 #include "gguf.h"
 
+#include <algorithm>
 #include <cassert>
 #include <cmath>
 #include <cstdlib>
 #include <cstring>
 #include <fstream>
 #include <map>
+#include <memory>
```

Comment:
some of these are already included by `clip-impl.h` - do we really need to include them again here?